scripts/openstack-fake-update.sh

```diff
@@ -0,0 +1,116 @@
+
+if [ -z "$OPERATOR_NAMESPACE" ]; then
+    echo "Please set OPERATOR_NAMESPACE"; exit 1
+fi
+if [ -z "$NAMESPACE" ]; then
+    echo "Please set NAMESPACE"; exit 1
+fi
+
+OPENSTACK_OPERATOR_CSV=$(oc get csv -n $OPERATOR_NAMESPACE -o name | grep openstack-operator)
+oc get $OPENSTACK_OPERATOR_CSV -o yaml -n $OPERATOR_NAMESPACE  > $OUTFILE
+sed -i $OUTFILE -e "s|value: .*/podified-antelope-centos9/\(.*\)@.*|value: quay.io/podified-antelope-centos9/\1:current-podified|g"
+
+OPENSTACK_VERSION_CR=$(oc get openstackversion -n $NAMESPACE -o name)
+OPENSTACK_DEPLOYED_VERSION=$(oc get -n $NAMESPACE $OPENSTACK_VERSION_CR --template={{.spec.targetVersion}})
+sed -i $OUTFILE -e "s|value: $OPENSTACK_DEPLOYED_VERSION|value: $VERSION|"
+
+oc apply -f $OUTFILE
+
+# wait for openstack-operator to start reinstallation
+until $(oc get csv -l operators.coreos.com/openstack-operator.openstack-operators -n openstack-operators | grep -q Installing); do
+    sleep 1;
+done
+
+# wait for openstack-operator to installation to complete
+until $(oc get csv -l operators.coreos.com/openstack-operator.openstack-operators -n openstack-operators | grep -q Succeeded); do
+    sleep 1;
+done
+
+# wait until openstackversion cr completes reconcile, status.availableVersion should be the same as VERSION
+until $(oc get -n openstack $OPENSTACK_VERSION_CR  --template={{.status.availableVersion}} | grep -q $VERSION); do
+    sleep 1;
+done
+
+OPENSTACK_DEPLOYED_VERSION=$(oc get -n $NAMESPACE $OPENSTACK_VERSION_CR --template={{.spec.targetVersion}})
+
+cat <<EOF >openstackversionpatch.yaml
+    "spec": {
+      "targetVersion": "$VERSION"
+      }
+EOF
+
+oc patch $OPENSTACK_VERSION_CR  --type=merge  --patch-file openstackversionpatch.yaml
+
+# wait for ovn update on control plane
+oc wait $OPENSTACK_VERSION_CR --for=condition=MinorUpdateOVNControlplane --timeout=$TIMEOUT
+
+# start ovn update on data plane
+DATAPLANE_NODESET=$(oc get openstackdataplanenodeset -o name | awk -F'/' '{print $2}')
+DATAPLANE_DEPLOYMENT=$(oc get openstackdataplanedeployment -o name | awk -F'/' '{print $2}')
+
+cat <<EOF >edpm-deployment-ovn-update.yaml
+apiVersion: dataplane.openstack.org/v1beta1
+kind: OpenStackDataPlaneDeployment
+metadata:
+  name: $DATAPLANE_DEPLOYMENT-ovn-update
+spec:
+  nodeSets:
+    - $DATAPLANE_NODESET
+  servicesOverride:
+    - ovn
+EOF
+
+oc create -f edpm-deployment-ovn-update.yaml
+
+oc get openstackdataplanedeployment
+# wait for ovn dataplane completes
+oc wait $OPENSTACK_VERSION_CR  --for=condition=MinorUpdateOVNDataplane --timeout=$TIMEOUT
+echo "MinorUpdateOVNDataplane completed"
+
+# wait for control plane update to complete
+oc wait $OPENSTACK_VERSION_CR --for=condition=MinorUpdateControlplane --timeout=$TIMEOUT
+
+# TODO: fix docs to include step
+# create openstackdataplaneservice update
+
+oc apply -f https://raw.githubusercontent.com/openstack-k8s-operators/openstack-operator/main/config/services/dataplane_v1beta1_openstackdataplaneservice_update.yaml
```

Comment:
You may need to be able to use a pinned sha, rather than the file from the head of the branch. I suggest that you make room for that possibility.

Reply:
outdated, line removed after bz was fixed